### PR TITLE
Add new campaign path, including two new scenarios

### PR DESCRIPTION
**Pull request by MadMax3947**
Add new campaign path allowing player to give up Kogw in scenario 9 and retain Theracar for the rest of the campaign. Includes two new scenarios.

Of note, adds optional dependency to UMC_Music_Book_10; please add to pbl file `dependencies=UMC_Music_Book_10`

### CODE REVIEW

File: _editor.cfg
Comment (cooljeanius) [inline _editor.cfg:41]: why remove this?

File: _editor.cfg
Comment (cooljeanius) [inline _editor.cfg:56]: likewise, why remove this?

File: _main.cfg
Comment (cooljeanius) [inline _main.cfg]: why comment this out?

File: _main.cfg
Comment (cooljeanius) [inline _main.cfg]: likewise, why comment this out?

File: lua/new_side.lua
Comment (cooljeanius) [inline lua/new_side.lua]: Is there a bug opened about this against upstream Wesnoth that we could cross-reference?

File: lua/npc.lua
Comment (cooljeanius) [inline lua/npc.lua:116]: I'm wondering if this fixes (or at least addresses) issue #8 in this repo at all?

File: macros/special-notes.cfg
Comment (cooljeanius) [inline macros/special-notes.cfg]: So, I've been trying to expunge usage of `SPECIAL_NOTES` macros from this campaign, since they're deprecated; see issue #12 in this repo... adding new ones would be kind of moving backwards in that regards...

File: macros/special-notes.cfg
Comment (cooljeanius) [inline macros/special-notes.cfg]: why replace the fancy apostrophe with the plain kind?

File: _editor.cfg
Comment (cooljeanius) [reply _editor.cfg:41]: ah wait never mind, I see now, you changed it to pull from UMC Book 10 instead...

File: credits.cfg
Comment (cooljeanius) [inline credits.cfg]: If the campaign is being changed to use the UMC Book 10 copy of this instead, does it still need to be credited here as well?

File: macros/weapon-specials.cfg
Comment (cooljeanius) [inline macros/weapon-specials.cfg]: again, see issue #12

File: units/undead/Avatar_of_Ending.cfg
Comment (cooljeanius) [inline units/undead/Avatar_of_Ending.cfg]: why's this line commented out?

File: scenarios/15_Gate_Storms.cfg
Comment (cooljeanius) [inline scenarios/15_Gate_Storms.cfg:1]: I wonder why this one isn't getting detected as a rename? Did you forget to use `git mv` to do it?

**Comment by cooljeanius** [comment]
Hm, it looks like now that @czyzby has fixed czyzby/wesnoth-wml-linter#7, there are now some additional `wmllint` warnings appearing; I'll have to look into those...

File: _editor.cfg
Comment (cooljeanius) [reply _editor.cfg:56]: as with the previous one, never mind, I see now that you changed it to pull from UMC Book 10 instead...

**Comment by cooljeanius** [comment]
> Hm, it looks like now that @czyzby has fixed [czyzby/wesnoth-wml-linter#7](https://github.com/czyzby/wesnoth-wml-linter/issues/7), there are now some additional `wmllint` warnings appearing; I'll have to look into those...

...ok, I've got a new baseline for the linting workflow; could you try rebasing this PR on the latest commit from this branch?

**Comment by MadMax3947** [comment]
> > Hm, it looks like now that @czyzby has fixed [czyzby/wesnoth-wml-linter#7](https://github.com/czyzby/wesnoth-wml-linter/issues/7), there are now some additional `wmllint` warnings appearing; I'll have to look into those...
> 
> ...ok, I've got a new baseline for the linting workflow; could you try rebasing this PR on the latest commit from this branch?

Done!

File: units/undead/Avatar_of_Ending.cfg
Comment (MadMax3947) [reply units/undead/Avatar_of_Ending.cfg]: I think it was commented out in the original code that this was based on (from Dark Master project).

File: macros/special-notes.cfg
Comment (MadMax3947) [reply macros/special-notes.cfg]: For consistency's sake with most of the other apostrophes that were plain, but agree that fancy apostrophe looks better.

File: lua/npc.lua
Comment (MadMax3947) [reply lua/npc.lua:116]: Doesn't address #8 specifically, rather this allows the macro to be used with birds that aren't on an idle_ai side (otherwise AI would be free to use birds to attack).

File: _main.cfg
Comment (MadMax3947) [reply _main.cfg]: since set_experience isn't a mandatory key for modify_unit_type; if a future Wesnoth version changes the exp needed for advancement it doesn't need to be overridden.

File: macros/special-notes.cfg
Comment (MadMax3947) [reply macros/special-notes.cfg]: Fixed; should close #12.

File: scenarios/15_Gate_Storms.cfg
Comment (MadMax3947) [reply scenarios/15_Gate_Storms.cfg:1]: I did use `git mv`, not sure why it attributed it to a new file anyway.

File: credits.cfg
Comment (MadMax3947) [reply credits.cfg]: We do not - removed this line.

File: macros/special-notes.cfg
Comment (cooljeanius) [reply macros/special-notes.cfg]: ok, I guess that's fine.

File: scenarios/15_Gate_Storms.cfg
Comment (cooljeanius) [reply scenarios/15_Gate_Storms.cfg:1]: huh... oh well, I guess it's not important enough to hold up the PR over...

File: _main.cfg
Comment (cooljeanius) [reply _main.cfg]: I think it might be better to explicitly override it anyways, for the sake of stability and consistency across versions. If a future Wesnoth version changes the exp needed for advancement, and we *don't* override that amount, then we'd need to do some other sort of rebalancing instead...

File: lua/npc.lua
Comment (cooljeanius) [reply lua/npc.lua:116]: OK, got it.

File: macros/special-notes.cfg
Comment (cooljeanius) [reply macros/special-notes.cfg]: ...ok, I see, in 4e73846, got it.

File: macros/weapon-specials.cfg
Comment (cooljeanius) [reply macros/weapon-specials.cfg]: fixed by 4e73846.

File: units/undead/Avatar_of_Ending.cfg
Comment (cooljeanius) [reply units/undead/Avatar_of_Ending.cfg]: Personally I prefer that code either be ifdef-ed out (so that it can be re-enabled more easily), or just removed entirely, instead of commented out... which of the 2 do you think would make more sense here?

**Comment by cooljeanius** [comment]
let me try compressing some of those images myself...

File: images/portraits/SkeletonDragon52.png
Comment (cooljeanius) [inline images/portraits/SkeletonDragon52.png:1]: with the way it's cropped... I'm wondering if that looks weird when used in-game at all?

File: images/portraits/sheep.png
Comment (cooljeanius) [inline images/portraits/sheep.png:1]: I thought black-background portraits were deprecated and that we were all supposed to be using clear-background portraits now instead...

File: macros/story.cfg
Comment (cooljeanius) [inline macros/story.cfg:1]: I didn't look at this one too closely, but I'll just assume it's ok

**Comment by cooljeanius** [comment]
OK, done with my current review... one more thing I'd like to get down before merging: could you please confirm what sort of testing you've performed with these changes? I generally like to test before making releases, but if you can just confirm that you've done so yourself, that would be fine, too. 

> Of note, adds optional dependency to UMC_Music_Book_10; please add to pbl file `dependencies=UMC_Music_Book_10`

This reminds me, I want to get something set up like Iris has for IftU and AtS, where the pbl file is generated by a Makefile, so that the publicly-viewable parts of the pbl file can be checked into the repository, while still keeping the secret parts private...

File: units/undead/Avatar_of_Ending.cfg
Comment (MadMax3947) [reply units/undead/Avatar_of_Ending.cfg]: probably just remove it entirely for now

File: images/portraits/sheep.png
Comment (MadMax3947) [reply images/portraits/sheep.png:1]: all-black are absolutely deprecated; unfortunately the original (from a really old campaign Rebellion in the Fields) had the black background. Could honestly just remove the portrait altogether since sheep aren't actually speaking.

File: images/portraits/SkeletonDragon52.png
Comment (MadMax3947) [reply images/portraits/SkeletonDragon52.png:1]: The wings look a little weird, but of the different angles/crops at the original source (https://forums.wesnoth.org/viewtopic.php?p=682633) this looked the best to me.

File: units/undead/Avatar_of_Ending.cfg
Comment (cooljeanius) [reply units/undead/Avatar_of_Ending.cfg]:
```suggestion
```

File: _main.cfg
Comment (MadMax3947) [reply _main.cfg]: sure but I feel that could go either way, i.e. if a future Wesnoth version changes the exp and adjusts some other value to compensate, etc.

**Comment by MadMax3947** [comment]
> OK, done with my current review... one more thing I'd like to get down before merging: could you please confirm what sort of testing you've performed with these changes? I generally like to test before making releases, but if you can just confirm that you've done so yourself, that would be fine, too.
> 
> > Of note, adds optional dependency to UMC_Music_Book_10; please add to pbl file `dependencies=UMC_Music_Book_10`
> 
> This reminds me, I want to get something set up like Iris has for IftU and AtS, where the pbl file is generated by a Makefile, so that the publicly-viewable parts of the pbl file can be checked into the repository, while still keeping the secret parts private...

Have tested new scenarios to completion as well as new events in other scenarios. May need some ongoing balancing though; haven't actually otherwise _played_ wesnoth in quite a while to have a perfect sense of this.

File: units/undead/Avatar_of_Ending.cfg
Comment (cooljeanius) [reply units/undead/Avatar_of_Ending.cfg]: ah ok I see that was included in [ac8aee5](https://github.com/cooljeanius/Flight_Freedom/pull/18/commits/ac8aee5e6fb3d1bf231b3f5053e383c7ef0e26ef) now...

File: images/portraits/SkeletonDragon52.png
Comment (cooljeanius) [reply images/portraits/SkeletonDragon52.png:1]: ok, I'll take your word for it

File: images/portraits/sheep.png
Comment (cooljeanius) [reply images/portraits/sheep.png:1]: Portraits are still visible in the in-game unit descriptions, and on `units.wesnoth.org`, though. Let me try turning the background transparent myself and see if that works... (unless you'd like to beat me to it...)

File: _main.cfg
Comment (cooljeanius) [reply _main.cfg]: Well ok, could we at least just switch it to ifdef-ing it out instead of commenting it out, then? I'm thinking like:
```suggestion
#ifdef FORCE_EXP_AMT
        set_experience=41
#endif
```

File: _main.cfg
Comment (cooljeanius) [reply _main.cfg]: As with my previous suggestion:
```suggestion
#ifdef FORCE_EXP_AMT
        set_experience=33
#endif
```

File: images/portraits/sheep.png
Comment (cooljeanius) [reply images/portraits/sheep.png:1]: OK, made transparent in [1900f31](https://github.com/cooljeanius/Flight_Freedom/pull/18/commits/1900f319b6ab71766b5a011af3ab1dd1b4796926).

**Comment by cooljeanius** [comment]
> Have tested new scenarios to completion as well as new events in other scenarios. May need some ongoing balancing though; haven't actually otherwise _played_ wesnoth in quite a while to have a perfect sense of this.

OK, well I guess I can get to that in a later release...

File: lua/new_side.lua
Comment (cooljeanius) [reply lua/new_side.lua]: @MadMax3947 any further info to share on this one?

Comment (MadMax3947) [reply lua/new_side.lua]: Just opened one, https://github.com/wesnoth/wesnoth/issues/8193

File: _main.cfg
Comment (MadMax3947) [reply _main.cfg]: Updated!

File: lua/new_side.lua
Comment (cooljeanius) [reply lua/new_side.lua]: ok, thanks

Comment (MadMax3947) [reply lua/new_side.lua]: so looking at the issues with `wesnoth.sides.create()` and given there's been talk about removing it, I implemented the revealed dunefolk side differently in https://github.com/cooljeanius/Flight_Freedom/pull/18/commits/eacd3d54026243361e89deb6f3665b4527a93d54

**Comment by MadMax3947** [comment]
> > Have tested new scenarios to completion as well as new events in other scenarios. May need some ongoing balancing though; haven't actually otherwise _played_ wesnoth in quite a while to have a perfect sense of this.
> 
> OK, well I guess I can get to that in a later release...

In that case I think this PR's probably ready to merge!

File: _main.cfg
Comment (cooljeanius) [reply _main.cfg]: ok, I see that was in f398700...

File: _main.cfg
Comment (cooljeanius) [reply _main.cfg]: fixed as of f398700.

**Review by cooljeanius — APPROVED** [review]
LGTM; I'll squash and then do some follow-up commits before releasing...

**Pull request merged**
